docs/docs/tutorials/contract_tutorials/token_contract.md

```diff
@@ -409,18 +449,22 @@ A getter function for checking the token `total_supply`.
 
 #include_code total_supply /noir-projects/noir-contracts/contracts/token_contract/src/main.nr rust
 
-#### `balance_of_private`
-
-A getter function for checking the private balance of the provided Aztec account. Note that the [Private Execution Environment (PXE)](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/yarn-project/pxe) must have `ivsk_app` ([app-specific secret key](../../aztec/concepts/accounts/keys.md#scoped-keys)) in order to decrypt the notes.
-
-#include_code balance_of_private /noir-projects/noir-contracts/contracts/token_contract/src/main.nr rust
-
 #### `balance_of_public`
 
 A getter function for checking the public balance of the provided Aztec account.
 
 #include_code balance_of_public /noir-projects/noir-contracts/contracts/token_contract/src/main.nr rust
 
+### Unconstrained function implementations
+
+Unconstrained functions are similar to `view` functions in Solidity in that they only return information from the contract storage or compute and return data without modifying contract storage. They are different from view functions in that the values are returned from the user's PXE and are not constrained by the contract's definition--if there is bad data in the user's PXE, they will get bad data back.
+
+#### `balance_of_private`
+
+A getter function for checking the private balance of the provided Aztec account. Note that the [Private Execution Environment (PXE)](https://github.com/AztecProtocol/aztec-packages/tree/#include_aztec_version/yarn-project/pxe) must have `ivsk_app` ([app-specific secret key](../../aztec/concepts/accounts/keys.md#scoped-keys)) in order to decrypt the notes.
```

Comment:
nsk_app is an app-specific secret key. Should ivsk_app have a different name and link to "incoming-viewing-keys" section instead of "scoped-keys" section?

Reply:
updated the link to the appropriate section.

`nsk_app` is the app specific nullifier secret key. are you suggesting we change what the `ivsk_app` is called? I think that would be a larger discussion with mike and others working on keys.